<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>format &mdash; snptools version 1.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=5c1c9e08"></script>
        <script src="_static/doctools.js?v=888ff710"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="tests.plink package" href="format.tests.html" />
    <link rel="prev" title="finalreport.tests.test_finalreport module" href="finalreport.tests.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            snptools
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="description.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="description.html#current-features">Current Features</a></li>
<li class="toctree-l1"><a class="reference internal" href="description.html#the-dataexplore-application">The DataExplore application</a></li>
<li class="toctree-l1"><a class="reference internal" href="description.html#links">Links</a></li>
<li class="toctree-l1"><a class="reference internal" href="description.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="description.html#installation">Installation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="modules.html">snptools</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="finalreport.html">finalreport</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">format</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#module-format">format package</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#format.Snp"><code class="docutils literal notranslate"><span class="pre">Snp</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#format.make_fam"><code class="docutils literal notranslate"><span class="pre">make_fam()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#format.make_lgen"><code class="docutils literal notranslate"><span class="pre">make_lgen()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#format.make_map"><code class="docutils literal notranslate"><span class="pre">make_map()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#format.make_ped"><code class="docutils literal notranslate"><span class="pre">make_ped()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#tests-package">tests package</a><ul>
<li class="toctree-l4"><a class="reference internal" href="format.tests.html">tests.plink package</a></li>
<li class="toctree-l4"><a class="reference internal" href="format.tests.html#tests-snp-package">tests.snp package</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="parentage.html">parentage</a></li>
<li class="toctree-l2"><a class="reference internal" href="stat.html">stat package</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">snptools</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="modules.html">snptools</a></li>
      <li class="breadcrumb-item active">format</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/format.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="format">
<h1>format<a class="headerlink" href="#format" title="Link to this heading"></a></h1>
<section id="module-format">
<span id="format-package"></span><h2>format package<a class="headerlink" href="#module-format" title="Link to this heading"></a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="format.Snp">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">format.</span></span><span class="sig-name descname"><span class="pre">Snp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">fmt</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'uga'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="_modules/format/_snp.html#Snp"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.Snp" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>The process of converting genomic map data - FinalReport.txt obtained
from Illumin. Recoding allele data into quantitative data, saving in the
format necessary for calculating gblup on blupf90.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="format.Snp._ALLELE_CODE">
<span class="sig-name descname"><span class="pre">_ALLELE_CODE</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'--':</span> <span class="pre">5,</span> <span class="pre">'AA':</span> <span class="pre">0,</span> <span class="pre">'AB':</span> <span class="pre">1,</span> <span class="pre">'BA':</span> <span class="pre">1,</span> <span class="pre">'BB':</span> <span class="pre">2}</span></em><a class="headerlink" href="#format.Snp._ALLELE_CODE" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="format.Snp._FIELDS">
<span class="sig-name descname"><span class="pre">_FIELDS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['SNP_NAME',</span> <span class="pre">'SAMPLE_ID',</span> <span class="pre">'SNP']</span></em><a class="headerlink" href="#format.Snp._FIELDS" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="format.Snp._F_DTYPE">
<span class="sig-name descname"><span class="pre">_F_DTYPE</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'SAMPLE_ID':</span> <span class="pre">&lt;class</span> <span class="pre">'str'&gt;,</span> <span class="pre">'SNP':</span> <span class="pre">&lt;class</span> <span class="pre">'str'&gt;,</span> <span class="pre">'SNP_NAME':</span> <span class="pre">&lt;class</span> <span class="pre">'str'&gt;}</span></em><a class="headerlink" href="#format.Snp._F_DTYPE" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="format.Snp._add_space">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">_add_space</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_len</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="_modules/format/_snp.html#Snp._add_space"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.Snp._add_space" title="Link to this definition"></a></dt>
<dd><p>Adding spaces up to the maximum length of the value in the
sample_id data</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>value</strong> – <ul>
<li><p>Sample_id value</p></li>
</ul>
</p></li>
<li><p><strong>max_len</strong> – <ul>
<li><p>Max len sample_id value</p></li>
</ul>
</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>Return replacing value</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="format.Snp._format_uga">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">_format_uga</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="_modules/format/_snp.html#Snp._format_uga"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.Snp._format_uga" title="Link to this definition"></a></dt>
<dd><p>Data format to use snp in plink and blupf90</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="format.Snp.data">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">data</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">DataFrame</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></em><a class="headerlink" href="#format.Snp.data" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="format.Snp.process">
<span class="sig-name descname"><span class="pre">process</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="_modules/format/_snp.html#Snp.process"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.Snp.process" title="Link to this definition"></a></dt>
<dd><p>Data processing and formatting. Calculation of statistical
information</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>data</strong> – <ul class="simple">
<li><p>Data from FinalReport file. Example:</p></li>
</ul>
<p>SNP Name  Sample ID  Allele1 - AB  Allele2 - AB  GC Score  GT Score
ABCA12  14814   A       A       0.4048  0.8164
ARS-BFGL-BAC-13031      14814   B       B       0.9083  0.8712
ARS-BFGL-BAC-13039      14814   A       A       0.9005  0.9096
ARS-BFGL-BAC-13049      14814   A       B       0.9295  0.8926</p>
</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>Returns true if the data was formatted successfully and</p></li>
</ul>
</p>
</dd>
</dl>
<p>statistical information was calculated, false if an error.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="format.Snp.to_file">
<span class="sig-name descname"><span class="pre">to_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="_modules/format/_snp.html#Snp.to_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.Snp.to_file" title="Link to this definition"></a></dt>
<dd><p>Saving data to a file</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>file_path</strong> – <ul class="simple">
<li><p>Path to file</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="format.make_fam">
<span class="sig-prename descclassname"><span class="pre">format.</span></span><span class="sig-name descname"><span class="pre">make_fam</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sid_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fid_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">father_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mother_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sex_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sex_val</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pheno_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pheno_val</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">-9</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="_modules/format/_plink.html#make_fam"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.make_fam" title="Link to this definition"></a></dt>
<dd><p>PLINK sample information file
<a class="reference external" href="https://www.cog-genomics.org/plink/1.9/formats#fam">https://www.cog-genomics.org/plink/1.9/formats#fam</a></p>
<blockquote>
<div><p>A text file with no header line, and one line per sample with the</p>
</div></blockquote>
<p>following six fields:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Family ID (‘FID’)</p></li>
<li><p>Within-family ID (‘IID’; cannot be ‘0’)</p></li>
<li><p>Within-family ID of father (‘0’ if father isn’t in dataset)</p></li>
<li><p>Within-family ID of mother (‘0’ if mother isn’t in dataset)</p></li>
<li><p>Sex code (‘1’ = male, ‘2’ = female, ‘0’ = unknown)</p></li>
<li><dl class="simple">
<dt>Phenotype value (‘1’ = control, ‘2’ = case, ‘-9’/’0’/non-numeric =</dt><dd><p>missing data if case/control)</p>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> – <ul>
<li><p>Snp data that contain full or partial information on the</p></li>
</ul>
<p>animal</p>
</p></li>
<li><p><strong>fid_col</strong> – <ul>
<li><p>Family ID, default value “1”. Must not contain</p></li>
</ul>
<p>underline - “_”</p>
</p></li>
<li><p><strong>sid_col</strong> – <ul>
<li><p>Within-family ID (‘IID’; cannot be ‘0’). Must not contain</p></li>
</ul>
<p>underline - “_”</p>
</p></li>
<li><p><strong>father_col</strong> – <ul>
<li><p>Within-family ID of father (‘0’ if father isn’t in</p></li>
</ul>
<p>dataset)</p>
</p></li>
<li><p><strong>mother_col</strong> – <ul>
<li><p>Within-family ID of mother (‘0’ if mother isn’t in</p></li>
</ul>
<p>dataset)</p>
</p></li>
<li><p><strong>sex_col</strong> – <ul>
<li><p>Sex column name in data</p></li>
</ul>
</p></li>
<li><p><strong>sex_val</strong> – <ul>
<li><p>Sex code (‘1’ = male, ‘2’ = female, ‘0’ = unknown)</p></li>
</ul>
</p></li>
<li><p><strong>pheno_col</strong> – <ul>
<li><p>Pheno column name in data</p></li>
</ul>
</p></li>
<li><p><strong>pheno_val</strong> – <ul>
<li><p>Phenotype value (‘1’ = control, ‘2’ = case,</p></li>
</ul>
<p>’-9’/’0’/non-numeric = missing data if case/control)</p>
</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>Return data in formate .fam</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="format.make_lgen">
<span class="sig-prename descclassname"><span class="pre">format.</span></span><span class="sig-name descname"><span class="pre">make_lgen</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sid_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">snp_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alleles</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fid_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="_modules/format/_plink.html#make_lgen"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.make_lgen" title="Link to this definition"></a></dt>
<dd><p>PLINK long-format genotype file
<a class="reference external" href="https://www.cog-genomics.org/plink/1.9/formats#lgen">https://www.cog-genomics.org/plink/1.9/formats#lgen</a></p>
<blockquote>
<div><p>A text file with no header line, and one line per genotype call (or</p>
</div></blockquote>
<p>just not-homozygous-major calls if ‘lgen-ref’ was invoked) usually with
the following five fields:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Family ID</p></li>
<li><p>Within-family ID</p></li>
<li><p>Variant identifier</p></li>
<li><p>Allele call 1 (‘0’ for missing)</p></li>
<li><p>Allele call 2</p></li>
</ol>
<p>There are several variations which are also handled by PLINK; see the</p>
</div></blockquote>
<p>original discussion for details.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> – <ul>
<li><p>data the after parsing FinalReport.txt</p></li>
</ul>
</p></li>
<li><p><strong>sid_col</strong> – <ul>
<li></li>
</ul>
</p></li>
<li><p><strong>snp_name</strong> – <ul>
<li></li>
</ul>
</p></li>
<li><p><strong>fid_col</strong> – <ul>
<li><p>Family ID, default value “1”</p></li>
</ul>
</p></li>
<li><p><strong>alleles</strong> – <ul>
<li></li>
</ul>
</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>Return data in formate .lgen</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="format.make_map">
<span class="sig-prename descclassname"><span class="pre">format.</span></span><span class="sig-name descname"><span class="pre">make_map</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">manifest</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="_modules/format/_plink.html#make_map"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.make_map" title="Link to this definition"></a></dt>
<dd><p>PLINK text fileset variant information file
<a class="reference external" href="https://www.cog-genomics.org/plink/1.9/formats#map">https://www.cog-genomics.org/plink/1.9/formats#map</a></p>
<blockquote>
<div><p>A text file with no header line, and one line per variant with the</p>
</div></blockquote>
<p>following 3-4 fields:</p>
<blockquote>
<div><ol class="arabic simple">
<li><dl class="simple">
<dt>Chromosome code. PLINK 1.9 also permits contig names here, but most</dt><dd><p>older programs do not.</p>
</dd>
</dl>
</li>
<li><p>Variant identifier</p></li>
<li><dl class="simple">
<dt>Position in morgans or centimorgans (optional; also safe to use</dt><dd><p>dummy value of ‘0’)</p>
</dd>
</dl>
</li>
<li><p>Base-pair coordinate</p></li>
</ol>
<p>All lines must have the same number of columns (so either no lines</p>
</div></blockquote>
<p>contain the morgans/centimorgans column, or all of them do).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>manifest</strong> – <ul class="simple">
<li><p>The file that is taken on the Illumina website with</p></li>
</ul>
<p>full information about the chip</p>
</p>
</dd>
</dl>
<p><a class="reference external" href="https://support.illumina.com/downloads/bovinesnp50-v3-0-product-files.html">https://support.illumina.com/downloads/bovinesnp50-v3-0-product-files.html</a></p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><ul class="simple">
<li><p>Return data in formate .map</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="format.make_ped">
<span class="sig-prename descclassname"><span class="pre">format.</span></span><span class="sig-name descname"><span class="pre">make_ped</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sid_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">snp_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fid_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">father_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mother_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sex_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="_modules/format/_plink.html#make_ped"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#format.make_ped" title="Link to this definition"></a></dt>
<dd><p>Original standard text format for sample pedigree information and
genotype calls. Normally must be accompanied by a .map file.
<a class="reference external" href="https://www.cog-genomics.org/plink/1.9/formats#ped">https://www.cog-genomics.org/plink/1.9/formats#ped</a></p>
<blockquote>
<div><p>The PED file has 6 fixed columns at the beginning followed by the SNP</p>
</div></blockquote>
<p>information. The columns should be separated by a whitespace or a tab. The
first six columns hold the following information:</p>
<blockquote>
<div><ol class="arabic simple">
<li><dl class="simple">
<dt>Family ID (if unknown use the same id as for the sample id in</dt><dd><p>column two)</p>
</dd>
</dl>
</li>
<li><p>Sample ID</p></li>
<li><p>Paternal ID (if unknown use 0)</p></li>
<li><p>Maternal ID (if unknown use 0)</p></li>
<li><p>Sex (if unknown use 0)</p></li>
<li><p>Not used, set to 0</p></li>
<li><p>Rest of the columns: SNPs</p></li>
</ol>
<p>Here is a brief example of a genotype PED file containing 5 samples</p>
</div></blockquote>
<dl class="simple">
<dt>with 10 homozygous SNPs:</dt><dd><p>4304 4304 0 0 0 0 C C C C G G G G G G C C G G C C T T T T
6925 6925 0 0 0 0 C C C C T T G G A A C C G G C C T T T T
7319 7319 0 0 0 0 C C C C G G G G G G C C G G C C T T T T
6963 6963 0 0 0 0 A A C C T T G G A A C C G G C C T T T T
6968 6968 0 0 0 0 C C C C G G G G G G G G G G C C T T T T</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> – <ul>
<li><p>Snp data that contain full or partial information on the</p></li>
</ul>
<p>animal</p>
</p></li>
<li><p><strong>sid_col</strong> – <ul>
<li><p>Sample ID. Column name in data</p></li>
</ul>
</p></li>
<li><p><strong>snp_col</strong> – <ul>
<li><p>Snp column name in data</p></li>
</ul>
</p></li>
<li><p><strong>fid_col</strong> – <ul>
<li><p>Family ID column name in data (if unknown use the same</p></li>
</ul>
<p>id as for the sample id in column two)</p>
</p></li>
<li><p><strong>father_col</strong> – <ul>
<li><p>Paternal ID column name in data (if unknown use 0)</p></li>
</ul>
</p></li>
<li><p><strong>mother_col</strong> – <ul>
<li><p>Maternal ID column name in data (if unknown use 0)</p></li>
</ul>
</p></li>
<li><p><strong>sex_col</strong> – <ul>
<li><p>Sex column name in data (if unknown use 0)</p></li>
</ul>
</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p>Returns an array of data in ped format to work with the plink</p></li>
</ul>
<p>program</p>
</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="tests-package">
<h2>tests package<a class="headerlink" href="#tests-package" title="Link to this heading"></a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="format.tests.html">tests.plink package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="format.tests.html#module-format.tests.test_plink_fam">format.tests.test_plink_fam module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_fam.TestPlinkFormatPed"><code class="docutils literal notranslate"><span class="pre">TestPlinkFormatPed</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_fam.data_fam"><code class="docutils literal notranslate"><span class="pre">data_fam()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="format.tests.html#module-format.tests.test_plink_lgen">format.tests.test_plink_lgen module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_lgen.TestPlinkFormatLgen"><code class="docutils literal notranslate"><span class="pre">TestPlinkFormatLgen</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_lgen.data_lgen"><code class="docutils literal notranslate"><span class="pre">data_lgen()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="format.tests.html#module-format.tests.test_plink_map">format.tests.test_plink_map module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_map.TestPlinkFormatMap"><code class="docutils literal notranslate"><span class="pre">TestPlinkFormatMap</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_map.data_map"><code class="docutils literal notranslate"><span class="pre">data_map()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="format.tests.html#module-format.tests.test_plink_ped">format.tests.test_plink_ped module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_ped.TestPlinkFormatPed"><code class="docutils literal notranslate"><span class="pre">TestPlinkFormatPed</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_plink_ped.data_ped"><code class="docutils literal notranslate"><span class="pre">data_ped()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="format.tests.html#tests-snp-package">tests.snp package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="format.tests.html#module-format.tests.test_snp">format.tests.test_snp module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_snp.TestSNP"><code class="docutils literal notranslate"><span class="pre">TestSNP</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_snp.data_fr"><code class="docutils literal notranslate"><span class="pre">data_fr()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="format.tests.html#format.tests.test_snp.obj_snp"><code class="docutils literal notranslate"><span class="pre">obj_snp()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="finalreport.tests.html" class="btn btn-neutral float-left" title="finalreport.tests.test_finalreport module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="format.tests.html" class="btn btn-neutral float-right" title="tests.plink package" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Igor Locshinin.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>