Search.setIndex({"docnames": ["examples", "index", "install", "intro", "modules", "snplib", "snplib.finalreport", "snplib.format", "snplib.parentage", "snplib.statistics", "usage"], "filenames": ["examples.rst", "index.rst", "install.rst", "intro.rst", "modules.rst", "snplib.rst", "snplib.finalreport.rst", "snplib.format.rst", "snplib.parentage.rst", "snplib.statistics.rst", "usage.rst"], "titles": ["Code Examples", "Welcome to snplib\u2019s documentation!", "Installation", "Introduction", "snplib", "Packages", "Finalreport package", "Format package", "Parentage package", "Statistics package", "Usage"], "terms": {"header": [0, 6, 7], "gsgt": [0, 6], "version": [0, 6], "2": [0, 6, 7, 9], "0": [0, 6, 7, 9], "4": [0, 6, 7], "date": [0, 6], "1": [0, 6, 7, 9], "16": 0, "2023": 0, "9": [0, 7], "19": 0, "am": 0, "content": [0, 6], "ggp_hdv3_c": 0, "bpm": [0, 6], "num": [0, 6], "139376": 0, "total": [0, 6], "140668": 0, "sampl": [0, 6, 7, 8], "25": 0, "26": 0, "name": [0, 6, 7, 8, 9], "id": [0, 6, 7, 9], "allele1": [0, 6, 7], "forward": [0, 6], "allele2": [0, 6, 7], "top": [0, 6], "ab": [0, 6, 7, 9], "gc": [0, 6, 7], "score": [0, 6, 7], "x": 0, "y": 0, "ar": [0, 5, 7, 9, 10], "bfgl": [0, 7], "bac": [0, 7], "10172": 0, "ho840m003135245650": 0, "g": [0, 7], "b": [0, 6, 7], "9420": 0, "069": 0, "801": 0, "1020": 0, "9489": 0, "033": 0, "700": 0, "10245": 0, "c": [0, 7], "7277": 0, "152": 0, "504": 0, "10345": 0, "A": [0, 6, 7, 9], "9411": 0, "598": 0, "572": 0, "10375": 0, "9348": 0, "430": 0, "494": 0, "from": [0, 2, 6, 7, 9, 10], "snplib": [0, 2, 6, 7, 8, 9, 10], "import": [0, 2, 10], "obj_report": 0, "handl": [0, 6, 7], "path": [0, 6, 7], "snp_data": [0, 6], "output": 0, "datafram": [0, 3, 6, 7, 8, 9], "select": [0, 6], "alleles_ab": 0, "data_ab": 0, "alleles_top": 0, "data_top": 0, "alleles_forward": 0, "data_forward": 0, "494f": 0, "after": [0, 7], "raw": 0, "further": [0, 6], "analysi": 0, "sever": [0, 7], "step": [0, 2], "singl": [0, 3, 6], "nucleotid": [0, 3, 6], "polymorph": [0, 3, 6], "necessari": [0, 7], "The": [0, 1, 6, 7, 8, 9], "receiv": 0, "often": 0, "requir": [0, 2], "bring": 0, "standard": [0, 7], "form": [0, 6], "propos": 0, "modul": [0, 1, 5, 6, 7, 9], "includ": [0, 3, 5], "program": [0, 7], "gblup": [0, 7], "ssgblup": 0, "gwa": [0, 7], "input": 0, "obtain": [0, 7], "us": [0, 2, 3, 6, 7, 9], "genom": [0, 7], "evalu": 0, "i": [0, 1, 3, 6, 7, 8, 9], "uga": [0, 7], "panda": 0, "pd": 0, "data_finalreport": 0, "read_csv": 0, "sep": [0, 6], "t": [0, 6, 7], "obj": 0, "fmt": [0, 7], "obj_snp": 0, "to_fil": [0, 7], "sample_id": [0, 7], "14814": [0, 7], "02011015010000500": 0, "14815": 0, "01110152120222512": 0, "default": [0, 6, 7], "result": 0, "snp_name": [0, 7], "abca12": [0, 6, 7], "13031": [0, 7], "13039": [0, 7], "3": [0, 6, 7, 9], "13049": [0, 7], "17": 0, "18": 0, "20": 0, "thi": [0, 9], "page": [0, 1], "describ": 0, "special": 0, "which": [0, 6, 7, 9], "identifi": [0, 7], "extens": 0, "http": [0, 2, 7, 8, 9], "www": [0, 7, 8, 9], "cog": [0, 7], "org": [0, 2, 7, 8], "\u0440\u0430\u0441\u043f\u0440\u043e\u0441\u0442\u0440\u0430\u043d\u043d\u0435\u043d\u043d\u044b\u0435": 0, "\u0444\u043e\u043c\u0440\u0430\u0442\u044b": 0, "\u0434\u043b\u044f": 0, "\u043f\u0440\u043e\u0432\u0435\u0434\u0435\u043d\u0438\u044f": 0, "\u0430\u043d\u0430\u043b\u0438\u0437\u0430": 0, "ped": [0, 7], "map": [0, 7], "fam": [0, 7], "lgen": [0, 7], "make_map": [0, 5, 7, 10], "input_data": 0, "dir_fil": 0, "file_bovinesnp50": 0, "csv": 0, "data_map": 0, "view": 0, "chr": 0, "morgan": [0, 7], "mapinfo": 0, "bovinehd0100037694": 0, "bovinehd0100037699": 0, "bovinehd0100037703": 0, "bovinehd0100037704": 0, "taken": [0, 7], "illumina": [0, 7], "websit": [0, 7], "full": [0, 7], "inform": [0, 6, 7, 9], "about": [0, 7], "chip": [0, 7, 9], "support": [0, 7], "com": [0, 2, 7, 9], "download": [0, 7], "bovinesnp50": [0, 7], "v3": [0, 7], "product": [0, 7], "html": [0, 7], "make_p": [0, 5, 7, 10], "data_p": 0, "fid_col": [0, 7], "family_id": 0, "father_col": [0, 7], "father": [0, 7], "mother_col": [0, 7], "mother": [0, 7], "sex_col": [0, 7], "sex": [0, 7], "1100": 0, "1101": 0, "1102": 0, "1103": 0, "5": [0, 7, 9], "6": [0, 7], "7": 0, "8": 0, "fid": [0, 7], "sid": 0, "not_us": 0, "make_fam": [0, 5, 7, 10], "data_fam": 0, "pheno_col": [0, 7], "pheno": [0, 7], "025": 0, "022": 0, "052": 0, "12": 0, "13": 0, "14": [0, 6], "15": 0, "make_lgen": [0, 5, 7, 10], "data_lgen": 0, "gt": [0, 6, 7], "107232207": 0, "4048": [0, 6, 7], "8164": [0, 6, 7], "9083": [0, 7], "8712": [0, 7], "9005": [0, 7], "9096": [0, 7], "9295": [0, 7], "8926": [0, 7], "107237284": 0, "9566": 0, "9257": 0, "3098": 0, "8555": 0, "8613": 0, "8319": 0, "13059": 0, "poor": 0, "qualiti": 0, "uninform": 0, "can": [0, 2, 3], "exclud": [0, 9], "help": 0, "reduc": 0, "nois": 0, "improv": 0, "accuraci": 0, "given": [0, 9], "defin": [0, 9], "proport": [0, 9], "individu": [0, 9], "studi": [0, 9], "correspond": [0, 9], "miss": [0, 3, 7, 9], "In": [0, 9], "follow": [0, 2, 3, 7, 9], "we": [0, 9], "filter": [0, 6, 9], "95": [0, 9], "mean": [0, 9], "retain": [0, 9], "less": [0, 9], "than": [0, 9], "marker": [0, 7, 8, 9], "Of": [0, 9], "sai": [0, 9], "54k": [0, 9], "50k": [0, 9], "have": [0, 7, 9], "been": [0, 9], "genotyp": [0, 3, 7, 9], "particular": [0, 9], "anim": [0, 6, 7, 9], "93": [0, 9], "in_data": 0, "apaf1": [0, 6], "1104": 0, "1105": 0, "1106": 0, "1107": 0, "1108": 0, "1109": 0, "1110": 0, "call_rat": [0, 5, 9, 10], "id_col": [0, 9], "snp_col": [0, 7, 9], "727273": 0, "909091": 0, "818182": 0, "900": [0, 9], "cl635944_160": [0, 9], "how": [0, 9], "mani": [0, 9], "actual": [0, 9], "successfulli": [0, 7, 9], "read": [0, 6, 9], "assum": [0, 9], "600": [0, 9], "67": [0, 9], "13086": 0, "13093": 0, "13110": 0, "13111": 0, "13113": 0, "15633": 0, "15634": 0, "15637": 0, "15659": 0, "15668": 0, "15708": 0, "15718": 0, "data_df": 0, "882353": 0, "repres": [0, 3, 9], "incid": [0, 9], "gene": [0, 6, 9], "variant": [0, 6, 7, 9], "popul": [0, 9], "freq": [0, 9], "allele_freq": [0, 5, 9, 10], "seq_col": [0, 9], "000": 0, "889": 0, "minor": [0, 3, 9], "therefor": [0, 9], "occur": [0, 9], "within": [0, 7, 9], "maf": [0, 3], "minor_allele_freq": [0, 5, 9, 10], "22": 0, "principl": [0, 9], "state": [0, 9], "genet": [0, 3, 9], "variat": [0, 3, 7, 9], "remain": [0, 9], "constant": [0, 9], "one": [0, 7, 9], "gener": [0, 6, 9], "next": [0, 9], "absenc": [0, 9], "disturb": [0, 9], "factor": [0, 9], "natur": [0, 9], "scitabl": [0, 9], "definit": [0, 9], "122": [0, 9], "To": [0, 2], "test": [0, 9], "hypothesi": 0, "chi2": 0, "distribut": 0, "degre": [0, 9], "freedom": [0, 9], "hwe_test": [0, 5, 9, 10], "seq_snp": [0, 9], "seri": [0, 6, 8, 9], "list": [0, 5, 6, 7, 8], "int": [0, 7, 8, 9], "2212120": 0, "714": 0, "true": [0, 6, 7], "000000000102": 0, "125": 0, "fals": [0, 6, 7], "p": [0, 9], "valu": [0, 6, 7, 9], "here": [0, 7, 10], "hom1": 0, "10": [0, 6, 7], "het": 0, "500": 0, "hom2": 0, "5000": 0, "6515718999145375": 0, "onc": 0, "associ": 0, "pattern": 0, "relationship": 0, "between": 0, "phenotyp": [0, 7], "diseas": 0, "interest": 0, "icar": [0, 3, 8], "document": [0, 8], "genoex": [0, 8], "20guidelin": [0, 8], "20for": [0, 8], "20parentag": [0, 8], "20verif": [0, 8], "20and": [0, 8], "20discoveri": [0, 8], "20base": [0, 8], "20on": [0, 8], "20snp": [0, 8], "pdf": [0, 8], "isag": 0, "macer": 0, "found": 0, "see": [0, 7], "appendix": 0, "guidelin": 0, "04": 0, "dna": 0, "technologi": 0, "either": [0, 7, 9], "11": 0, "app": 0, "patern": [0, 7, 8], "accord": [0, 8], "recommend": [0, 8], "id41988163": 0, "id10512586": 0, "239": 0, "hapmap55441": 0, "rs29010990": 0, "240": 0, "hapmap59876": 0, "rs29018046": 0, "241": 0, "hapmap60017": 0, "rs29023471": 0, "242": 0, "ua": 0, "ifasa": 0, "5034": 0, "243": 0, "6532": 0, "isag_verif": [0, 5, 8], "obj_verif": 0, "isag_mark": [0, 8], "check_on": [0, 8], "descend": [0, 8], "parent": [0, 8], "snp_name_col": [0, 8], "print": 0, "num_conflict": [0, 8], "31": 0, "statu": [0, 8], "search": [0, 1, 8], "617": 0, "618": 0, "6154": 0, "619": 0, "620": 0, "8658": 0, "621": 0, "8833": 0, "isag_disc": [0, 5, 8], "obj_discoveri": 0, "search_par": [0, 8], "77": 0, "perc_conflict": [0, 8], "86": 0, "introduct": 1, "instal": 1, "easiest": 1, "wai": [1, 9], "anoth": [1, 6], "method": [1, 6, 9], "more": 1, "complic": 1, "usag": 1, "code": [1, 7], "exampl": [1, 6, 7, 9, 10], "head": 1, "file": [1, 3, 6, 7, 10], "finalreport": [1, 3, 4, 5, 7, 10], "process": [1, 3, 6, 7, 9, 10], "txt": [1, 2, 3, 6, 7], "prepar": [1, 3, 7, 10], "snp": [1, 3, 5, 6, 7, 8, 9, 10], "data": [1, 3, 6, 7, 8, 9, 10], "format": [1, 3, 4, 5, 9, 10], "statist": [1, 4, 5, 7, 10], "parentag": [1, 3, 4, 5, 10], "packag": [1, 4], "index": 1, "below": 2, "you": 2, "snptool": [2, 3, 10], "via": 2, "pip": 2, "pypi": 2, "project": 2, "clone": 2, "repositori": 2, "your": 2, "directori": 2, "git": 2, "github": [2, 9], "igorekloschinin": 2, "set": [2, 3, 8], "depend": 2, "cd": 2, "python": [2, 9], "m": 2, "r": 2, "tool": 3, "calcul": [3, 7, 9], "call": [3, 7, 9], "rate": [3, 9], "estim": 3, "analyz": 3, "provid": [3, 10], "work": [3, 7], "capabl": 3, "plink": [3, 7], "blupf90": [3, 7], "verif": [3, 5, 8, 10], "discoveri": [3, 5, 8, 10], "base": [3, 6, 7, 8], "percentag": 3, "allel": [3, 6, 7, 9], "frequenc": [3, 9], "hardi": [3, 9], "weinberg": [3, 9], "equilibrium": [3, 9], "librari": 5, "hwe": [5, 9, 10], "perform": [6, 7], "pars": [6, 7], "class": [6, 7, 8], "str": [6, 7, 8, 9], "none": [6, 7, 8, 9], "sourc": [6, 7, 8, 9], "object": [6, 7, 8], "contain": [6, 7], "paramet": [6, 7, 8, 9], "specif": 6, "site": 6, "whole": 6, "detect": 6, "locu": 6, "type": 6, "plu": 6, "design": 6, "delimit": 6, "2021": 6, "02": 6, "pm": 6, "bovinesnp50_v3_a1": 6, "53218": 6, "9067": 6, "9155": 6, "__pattern_data": 6, "re": 6, "compil": 6, "__pattern_head": 6, "__convert_s_id": 6, "path_fil": 6, "convert": [6, 7], "registr": 6, "number": [6, 7, 9], "xlsx": 6, "label": 6, "__handler_data": 6, "an": [6, 7], "arrai": [6, 7, 9], "__handler_head": 6, "meta": 6, "__sample_by_allel": 6, "field": [6, 7], "choos": 6, "keep": 6, "report": 6, "return": [6, 7, 9], "static": [6, 7], "_check_on_ru_symbol": 6, "seq": 6, "bool": [6, 9], "file_rep": 6, "conv_fil": 6, "highlight": 6, "wa": [6, 7], "success": 6, "were": [6, 9], "error": [6, 7, 9], "properti": [6, 7, 8], "dict": 6, "final_report": 6, "pointer": 6, "fail": 6, "certain": 7, "encod": 7, "illumin": 7, "recod": 7, "quantit": 7, "save": 7, "_allele_cod": 7, "aa": [7, 9], "ba": [7, 9], "bb": [7, 9], "_field": 7, "_f_dtype": 7, "_add_spac": 7, "max_len": 7, "ad": 7, "space": 7, "up": 7, "maximum": 7, "length": 7, "max": 7, "len": 7, "replac": 7, "_format_uga": 7, "file_path": 7, "sid_col": 7, "sex_val": 7, "pheno_v": 7, "text": 7, "line": 7, "per": 7, "six": 7, "famili": 7, "iid": 7, "cannot": 7, "isn": 7, "dataset": 7, "male": 7, "femal": 7, "unknown": 7, "control": 7, "case": 7, "non": 7, "numer": 7, "partial": 7, "must": 7, "underlin": 7, "_": 7, "column": [7, 8, 9], "long": 7, "just": 7, "homozyg": 7, "major": 7, "ref": 7, "invok": 7, "usual": 7, "five": 7, "There": 7, "also": 7, "origin": 7, "discuss": 7, "detail": 7, "manifest": 7, "fileset": 7, "chromosom": 7, "permit": 7, "contig": 7, "most": 7, "older": 7, "do": 7, "posit": 7, "centimorgan": 7, "option": 7, "safe": 7, "dummi": 7, "pair": 7, "coordin": 7, "all": 7, "same": 7, "so": 7, "them": 7, "pedigre": 7, "normal": 7, "accompani": 7, "ha": 7, "fix": [7, 8], "begin": 7, "should": 7, "separ": 7, "whitespac": 7, "tab": 7, "first": 7, "hold": 7, "two": 7, "matern": 7, "affect": 7, "unaffect": 7, "each": [7, 8, 9], "brief": 7, "4304": 7, "6925": 7, "7319": 7, "6963": 7, "6968": 7, "confirm": 8, "__status_defin": 8, "float": [8, 9], "discov": 8, "offer": 9, "callrat": 9, "occurr": 9, "ucg": 9, "pre": 9, "sequenc": 9, "transmit": 9, "pass": 9, "string": 9, "obs_het": 9, "obs_hom1": 9, "obs_hom2": 9, "interpret": 9, "jeremymcra": 9, "snphwe": 9, "observ": 9, "heterozygot": 9, "homozygotes1": 9, "homozygotes2": 9, "where": 9, "crit_chi2": 9, "841": 9, "critic": 9, "expect": 9, "other": 9, "df": 9, "84": 9, "05": 9, "decis": 9, "inspect": 9, "command": 10, "varieti": 10, "oper": 10, "comput": 10, "stat": 10}, "objects": {"snplib": [[6, 0, 0, "-", "finalreport"], [7, 0, 0, "-", "format"], [8, 0, 0, "-", "parentage"], [9, 0, 0, "-", "statistics"]], "snplib.finalreport": [[6, 1, 1, "", "FinalReport"]], "snplib.finalreport.FinalReport": [[6, 2, 1, "", "__PATTERN_DATA"], [6, 2, 1, "", "__PATTERN_HEADER"], [6, 3, 1, "", "__convert_s_id"], [6, 3, 1, "", "__handler_data"], [6, 3, 1, "", "__handler_header"], [6, 3, 1, "", "__sample_by_allele"], [6, 3, 1, "", "_check_on_ru_symbols"], [6, 3, 1, "", "handle"], [6, 4, 1, "", "header"], [6, 3, 1, "", "read"], [6, 4, 1, "", "snp_data"]], "snplib.format": [[7, 1, 1, "", "Snp"], [7, 5, 1, "", "make_fam"], [7, 5, 1, "", "make_lgen"], [7, 5, 1, "", "make_map"], [7, 5, 1, "", "make_ped"]], "snplib.format.Snp": [[7, 2, 1, "", "_ALLELE_CODE"], [7, 2, 1, "", "_FIELDS"], [7, 2, 1, "", "_F_DTYPE"], [7, 3, 1, "", "_add_space"], [7, 3, 1, "", "_format_uga"], [7, 4, 1, "", "data"], [7, 3, 1, "", "process"], [7, 3, 1, "", "to_file"]], "snplib.parentage": [[8, 1, 1, "", "Discovery"], [8, 1, 1, "", "Verification"], [8, 5, 1, "", "isag_disc"], [8, 5, 1, "", "isag_verif"]], "snplib.parentage.Discovery": [[8, 3, 1, "", "__status_define"], [8, 4, 1, "", "num_conflicts"], [8, 4, 1, "", "perc_conflicts"], [8, 3, 1, "", "search_parent"], [8, 4, 1, "", "status"]], "snplib.parentage.Verification": [[8, 3, 1, "", "check_on"], [8, 4, 1, "", "num_conflicts"], [8, 4, 1, "", "status"]], "snplib.statistics": [[9, 5, 1, "", "allele_freq"], [9, 5, 1, "", "call_rate"], [9, 5, 1, "", "hwe"], [9, 5, 1, "", "hwe_test"], [9, 5, 1, "", "minor_allele_freq"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:attribute", "3": "py:method", "4": "py:property", "5": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "attribute", "Python attribute"], "3": ["py", "method", "Python method"], "4": ["py", "property", "Python property"], "5": ["py", "function", "Python function"]}, "titleterms": {"code": 0, "exampl": 0, "head": 0, "file": 0, "finalreport": [0, 6], "process": 0, "txt": 0, "prepar": 0, "snp": 0, "data": 0, "format": [0, 7], "blupf90": 0, "plink": 0, "statist": [0, 9], "call": 0, "rate": 0, "frequenc": 0, "allel": 0, "hwe": 0, "hardi": 0, "weinberg": 0, "equilibrium": 0, "parentag": [0, 8], "verif": 0, "discoveri": 0, "welcom": 1, "snplib": [1, 4], "": 1, "document": 1, "indic": 1, "tabl": 1, "instal": 2, "The": 2, "easiest": 2, "wai": 2, "anoth": 2, "method": 2, "i": 2, "more": 2, "complic": 2, "introduct": 3, "packag": [5, 6, 7, 8, 9], "usag": 10}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.todo": 2, "sphinx.ext.viewcode": 1, "sphinx": 60}, "alltitles": {"Code Examples": [[0, "code-examples"]], "Head file FinalReport": [[0, "head-file-finalreport"]], "Processing the Finalreport.txt": [[0, "processing-the-finalreport-txt"]], "Preparation SNP files": [[0, "preparation-snp-files"]], "Data formatting": [[0, "data-formatting"]], "blupf90 format": [[0, "blupf90-format"]], "plink format": [[0, "plink-format"]], "Statistics": [[0, "statistics"]], "Call Rate": [[0, "call-rate"]], "Frequence Allele": [[0, "frequence-allele"]], "HWE (Hardy-Weinberg equilibrium)": [[0, "hwe-hardy-weinberg-equilibrium"]], "Parentage": [[0, "parentage"]], "Verification": [[0, "verification"]], "Discovery": [[0, "discovery"]], "Welcome to snplib\u2019s documentation!": [[1, "welcome-to-snplib-s-documentation"]], "Indices and tables": [[1, "indices-and-tables"]], "Installation": [[2, "installation"]], "The easiest way to install:": [[2, "the-easiest-way-to-install"]], "Another method is more complicated:": [[2, "another-method-is-more-complicated"]], "Introduction": [[3, "introduction"]], "snplib": [[4, "snplib"]], "Packages": [[5, "packages"]], "Finalreport package": [[6, "finalreport-package"]], "Format package": [[7, "format-package"]], "Parentage package": [[8, "parentage-package"]], "Statistics package": [[9, "statistics-package"]], "Usage": [[10, "usage"]]}, "indexentries": {"finalreport (class in snplib.finalreport)": [[6, "snplib.finalreport.FinalReport"]], "__pattern_data (snplib.finalreport.finalreport attribute)": [[6, "snplib.finalreport.FinalReport.__PATTERN_DATA"]], "__pattern_header (snplib.finalreport.finalreport attribute)": [[6, "snplib.finalreport.FinalReport.__PATTERN_HEADER"]], "__convert_s_id() (snplib.finalreport.finalreport method)": [[6, "snplib.finalreport.FinalReport.__convert_s_id"]], "__handler_data() (snplib.finalreport.finalreport method)": [[6, "snplib.finalreport.FinalReport.__handler_data"]], "__handler_header() (snplib.finalreport.finalreport method)": [[6, "snplib.finalreport.FinalReport.__handler_header"]], "__sample_by_allele() (snplib.finalreport.finalreport method)": [[6, "snplib.finalreport.FinalReport.__sample_by_allele"]], "_check_on_ru_symbols() (snplib.finalreport.finalreport static method)": [[6, "snplib.finalreport.FinalReport._check_on_ru_symbols"]], "handle() (snplib.finalreport.finalreport method)": [[6, "snplib.finalreport.FinalReport.handle"]], "header (snplib.finalreport.finalreport property)": [[6, "snplib.finalreport.FinalReport.header"]], "module": [[6, "module-snplib.finalreport"], [7, "module-snplib.format"], [8, "module-snplib.parentage"], [9, "module-snplib.statistics"]], "read() (snplib.finalreport.finalreport method)": [[6, "snplib.finalreport.FinalReport.read"]], "snp_data (snplib.finalreport.finalreport property)": [[6, "snplib.finalreport.FinalReport.snp_data"]], "snplib.finalreport": [[6, "module-snplib.finalreport"]], "snp (class in snplib.format)": [[7, "snplib.format.Snp"]], "_allele_code (snplib.format.snp attribute)": [[7, "snplib.format.Snp._ALLELE_CODE"]], "_fields (snplib.format.snp attribute)": [[7, "snplib.format.Snp._FIELDS"]], "_f_dtype (snplib.format.snp attribute)": [[7, "snplib.format.Snp._F_DTYPE"]], "_add_space() (snplib.format.snp static method)": [[7, "snplib.format.Snp._add_space"]], "_format_uga() (snplib.format.snp static method)": [[7, "snplib.format.Snp._format_uga"]], "data (snplib.format.snp property)": [[7, "snplib.format.Snp.data"]], "make_fam() (in module snplib.format)": [[7, "snplib.format.make_fam"]], "make_lgen() (in module snplib.format)": [[7, "snplib.format.make_lgen"]], "make_map() (in module snplib.format)": [[7, "snplib.format.make_map"]], "make_ped() (in module snplib.format)": [[7, "snplib.format.make_ped"]], "process() (snplib.format.snp method)": [[7, "snplib.format.Snp.process"]], "snplib.format": [[7, "module-snplib.format"]], "to_file() (snplib.format.snp method)": [[7, "snplib.format.Snp.to_file"]], "discovery (class in snplib.parentage)": [[8, "snplib.parentage.Discovery"]], "verification (class in snplib.parentage)": [[8, "snplib.parentage.Verification"]], "__status_define() (snplib.parentage.discovery method)": [[8, "snplib.parentage.Discovery.__status_define"]], "check_on() (snplib.parentage.verification method)": [[8, "snplib.parentage.Verification.check_on"]], "isag_disc() (in module snplib.parentage)": [[8, "snplib.parentage.isag_disc"]], "isag_verif() (in module snplib.parentage)": [[8, "snplib.parentage.isag_verif"]], "num_conflicts (snplib.parentage.discovery property)": [[8, "snplib.parentage.Discovery.num_conflicts"]], "num_conflicts (snplib.parentage.verification property)": [[8, "snplib.parentage.Verification.num_conflicts"]], "perc_conflicts (snplib.parentage.discovery property)": [[8, "snplib.parentage.Discovery.perc_conflicts"]], "search_parent() (snplib.parentage.discovery method)": [[8, "snplib.parentage.Discovery.search_parent"]], "snplib.parentage": [[8, "module-snplib.parentage"]], "status (snplib.parentage.discovery property)": [[8, "snplib.parentage.Discovery.status"]], "status (snplib.parentage.verification property)": [[8, "snplib.parentage.Verification.status"]], "allele_freq() (in module snplib.statistics)": [[9, "snplib.statistics.allele_freq"]], "call_rate() (in module snplib.statistics)": [[9, "snplib.statistics.call_rate"]], "hwe() (in module snplib.statistics)": [[9, "snplib.statistics.hwe"]], "hwe_test() (in module snplib.statistics)": [[9, "snplib.statistics.hwe_test"]], "minor_allele_freq() (in module snplib.statistics)": [[9, "snplib.statistics.minor_allele_freq"]], "snplib.statistics": [[9, "module-snplib.statistics"]]}})